bootstrap/pkg/apis/apps/ksapp/v1alpha1/application_types.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2018 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"fmt"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"math/rand"
+)
+
+const (
+	KsName    = "ks_app"
+	KsEnvName = "default"
+)
+
+func QuoteItems(items []string) []string {
+	withQuotes := []string{}
+	for _, item := range items {
+		withQuote := "\"" + item + "\""
+		withQuotes = append(withQuotes, withQuote)
+	}
+	return withQuotes
+}
+
+func RemoveItem(defaults []string, name string) []string {
+	pkgs := []string{}
+	for _, pkg := range defaults {
+		if pkg != name {
+			pkgs = append(pkgs, pkg)
+		}
+	}
+	return pkgs
+}
+
+func RemoveItems(defaults []string, names ...string) []string {
+	pkgs := make([]string, len(defaults))
+	copy(pkgs, defaults)
+	for _, name := range names {
+		pkgs = RemoveItem(pkgs, name)
+	}
+	return pkgs
+}
+
+var DefaultRegistry = &RegistryConfig{
+	Name: "kubeflow",
+	Repo: "https://github.com/kubeflow/kubeflow.git",
```

Comment:
User might hit Github API rate limit if they use default: always pull from github.
Shall we bake a release tag registry into our binary?
https://github.com/gobuffalo/packr

Or maybe follow up PRs? @jlewi

Reply:
Another option is keep current logic: curl and download release tag registry into tmp dir.
Pros: serve multi release tags with same go binary
Cons: doesn't work for environment without public network: on prem usage case

Reply:
@kunmingg i like baking in the release version as well as having versions kfctl could download. I think it would be best if we maintain default components, packages, parameters per version somewhere on github rather than having it has golang vars . @jlewi @ashahba thoughts?

Reply:
Definitely. we can have file like https://github.com/kubeflow/kubeflow/blob/master/components/gcp-click-to-deploy/app-config.yaml in github, shared as default to deploy.
And we can update the file as we cut new release so each release tag has its own components set.